Add `Common seeding methods` wiki page (#12843)
ppy/osu-wiki: wiki/Tournaments/Common_seeding_methods/en.md

## Official support

If you are seeking [official support](/wiki/Tournaments/Official_support) for your tournament and you are using the formulas or methods as described on this page exactly as they are written, you may refer to the name as described in the [seeding methods](#seeding-methods) section instead of specifying the exact formula. However, should there be any difference or change from the methods described here, you will be required to provide the full formula in your tournament's ruleset. If you are using a different seeding method not found on this page, make sure to specify the exact formulas and/or method used.

As always, a third party should be able to get the same results without needing additional information.

## Mini-glossary

The explanations below frequently use the following terms:

- A **team** is a single entity participating in the tournament. For example, the 6–8 players from each country in [OWC](/wiki/Tournaments/OWC) are considered one team, while each player in the [Lazer Grand Arena](/wiki/Tournaments/LGA/2024) is considered their own team.

> - A **score** is a numerical value used for determining a team's performance on a given map. For example, in a 3v3 team tournament, a score typically refers to the sum of the three team members' scores, rather than those three numbers separately. Some tournaments may allow teams multiple runs through a Qualifier mappool; in those cases, the team's score is often either the highest or the average of those runs.

Reviewer: ```suggestion
- A **score** is a numerical value used for determining a team's performance on a given map. For example, in a 3v3 team tournament, a score typically refers to the sum of the three team members' scores, rather than those three numbers separately. Some tournaments may allow teams multiple runs through a Qualifiers mappool; in those cases, the team's score is often either the highest or the average of those runs.
```
Consistency with naming

Would still like input on if "qualifier" should be lowercase when referring to *a* qualifier, rather than the name of a specific stage (which is the case with the majority of this page)

- A team's **seed** is a number indicating their overall ranking among all teams. Typically, after one of the seeding methods below is used, the top N teams (that is, the teams seeded 1, 2, ..., N) are placed into an elimination bracket or other competition format, with their locations being determined by their seeds.
- **Normalisation** is the process of dividing a team's score by any mod multipliers on the map. Typically, this is done to avoid artificial inflation of scores on modded maps.

## Seeding methods

<!-- NOTE FOR WIKI EDITORS: Please do not add or remove seeding methods without prior approval of the Tournament Committee -->

> Each subsection below explains, both in words and in a mathematical formula, a different seeding method commonly found in osu! tournaments. In all formulas, `m` refers to the number of maps in the Qualifier mappool.

Reviewer: ```suggestion
Each subsection below explains, both in words and in a mathematical formula, a different seeding method commonly found in osu! tournaments. In all formulas, `m` refers to the number of maps in the Qualifiers mappool.
```

### Average Score

This seeding method averages each team's scores on all maps. Typically, scores are first normalised. Teams are then seeded in order of this average, with the highest average receiving seed 1.

The average score for a team is given by

![Average Score formula](img/avgscore.png)

> If any of the maps are played with [Free Mod](/wiki/Gameplay/Game_modifier#free-mod), it must be explicitly specified whether each individual player's score on the map is normalised by the mods they use — effectively nullifying multipliers of any additional mods taken — or if there is an overall normalisation — where all scores are divided only by the globally required mod multipliers.

Reviewer 2: ```suggestion
If any of the maps are played with [Free Mod](/wiki/Gameplay/Game_modifier#free-mod), it must be explicitly specified whether each individual player's score on the map is normalised by the mods they use — effectively nullifying multipliers of any additional mods taken — or if there is an overall normalisation, where all scores are divided only by the globally required mod multipliers.
```

> ### Sum of Placements

Reviewer 2: we generally don't randomly declare proper nouns unless it pertain to spelling used in the game (e.g. mod names). other articles instead use bolding to introduce new terms

- Ziph's law (normally, only "Ziph" is the proper noun since it's a surname)
- Z-score (letters by themselves should be capitalised in any case)
- Z-sum
- average score
- sum of placements
- percent max
- percent difference

Note that tournaments in the past have often used the term "Z-sum" for what is instead called the "Z-percentile" seeding method below.[^Z-sum] The Z-sum seeding method should result in roughly half the teams having a negative total sum of Z-scores and half having a positive sum. The total sum over all maps should be exactly 0.

The Z-sum for a team is given by

![Z-sum formula](img/zsum.png)

Typically the sample standard deviation is used, rather than the population standard deviation.

### Z-percentile

This seeding method is almost identical to Z-sum, except that instead of assigning teams a Z-score, it assigns teams the corresponding percentile with respect to the normal distribution. In other words, Z-percentile estimates the percentage of teams beaten on each map by fitting scores to a bell curve, and each team will receive a value between 0 and 1 on each map played. Seeding is then determined by the sum of these values, with the highest sum receiving seed 1.

The Z-percentile sum for a team is given by

![Z-percentile formula](img/zpercentile.png)

> Here, `NORMCDF` is the cumulative distribution function for the standard normal distribution (also sometimes called `NORMSDIST` in software).

Reviewer 3: Maybe also worth linking the distribution itself?
```suggestion
Here, `NORMCDF` is the [cumulative distribution function](https://en.wikipedia.org/wiki/Cumulative_distribution_function) for the [standard normal distribution](https://en.wikipedia.org/wiki/Normal_distribution#Standard_normal_distribution) (also sometimes called `NORMSDIST` in software).
```

> Here, `NORMCDF` is the cumulative distribution function for the standard normal distribution (also sometimes called `NORMSDIST` in software).

Author: Added the standard normal distribution link in the line above where it is first mentioned.

### Zipf's law

Much like Sum of Placements, this seeding method ranks all scores on each map played, but it then gives the team points according to a version of the [Zipf–Mandelbrot law](https://en.wikipedia.org/wiki/Zipf%E2%80%93Mandelbrot_law). This yields a particularly high number of points for the top scorers on each map (largest gap between the first and second highest score), and seeding is then determined by the sum of all points received (with the highest sum receiving seed 1).

The usual formula used for assigning points is given by

![Zipf's law formula](img/zipf.png)

> Note that this seeding method may still make sense if `1.4 * m` is replaced by some other nonnegative real number, or if the numerator is replaced by another positive real number, or if the denominator is raised to some positive real number. But these changes must be specified explicitly if used.

Reviewer 3: yup, that's right!

### Weighting maps

Many of the methods above assign each team a number of points on each map and then add or average those points together to determine final seeding. It may also make sense to perform a weighted average so that some maps contribute more than others to final standings. For example, osu!mania tournaments may assign niche skill sets or physically demanding maps a lower weight than fundamental skill sets or accuracy-based challenges.

More precisely, a positive number for each map, called the `map weight`, should be specified. The final number used for seeding is then

![Weighting formula](img/weight.png)

> where `map points` is the quantity that the seeding method assigns to each map (e.g. `map placement` for Average Placement). Note that having all map weights be equal to `1` or `1/(number of maps)` would just be equivalent to summing or averaging the map points together.

Reviewer 3: Do we want to say "the sum of placements method" instead because that's what we called it earlier? It's a bit weird because we do say "Weighted average placement" below..

## Differences between methods

There are many factors that go into choosing a seeding method for a tournament. First of all, the following principles are important to keep in mind:

> - **Lack of Qualifiers**: All of the seeding methods above assume that teams all play through the same Qualifiers pool. If Qualifiers are not played, then it may make sense instead to seed by rank or [BWS](/wiki/Tournaments/Badge-weighted_seeding). (For example, a 2v2 tournament with team size 4 may seed teams by the sum of their two best BWS ranks, with the smallest sum being seeded 1st.)

Reviewer: ```suggestion
- **Lack of Qualifiers**: All of the seeding methods above assume that teams all play through the same Qualifiers mappool. If Qualifiers are not played, then it may make sense instead to seed by rank or [BWS](/wiki/Tournaments/Badge-weighted_seeding). (For example, a 2v2 tournament with team size 4 may seed teams by the sum of their two best BWS ranks, with the smallest sum being seeded 1st.)
```
consistency

  - Note that some tournaments may begin head-to-head competition by placing teams into a Group Stage or Swiss-system tournament instead of a standard elimination bracket. However, these other systems still require specification on how the teams are placed into matchups, so often some method of seeding is still required (either one of the ones mentioned in this page, or random drawings).
- **Scoring system**: Most of the seeding methods above are typically used when tournaments are played with the ScoreV2 scoring system. Thus, extra care should be taken when seeding tournaments that use ScoreV1, accuracy, or other methods of scoring. For example, a map played with ScoreV1 may have high outliers from well-performing teams, so it may be less appropriate to seed by Z-percentile or Percent Max.
- **Transparency and familiarity**: More complicated seeding methods, or new formulas that are not widely used in tournaments, may be more difficult for players and staff to understand or double-check. It may be advantageous to choose a seeding method that can be quickly sanity checked without needing heavy calculation.

### Comparing the most commonly used methods

> For reference, below is a list of Qualifier seeding methods used in the official osu! World Cups (which are team tournaments). Note that older tournaments often instead directly began with a Group Stage, choosing the qualifying teams by the osu! website's country rank.

Reviewer: ```suggestion
For reference, below is a list of Qualifiers seeding methods used in the official osu! World Cups (which are team tournaments). Note that older tournaments often instead directly began with a Group Stage, choosing the qualifying teams by the osu! website's country rank.
```

- If the highest score of 950,000 were instead much higher (like 1,200,000), seeding under Placement would remain the same, seeding under Percent Max would heavily deflate all other teams, and seeding under Z-percentile would be altered slightly (most notably in shifting the average and thus the range of scores in which improvements are most important).

Finally, a side-by-side comparison of all seeding methods on this page using the [osu! World Cup 2024](/wiki/Tournaments/OWC/2024) Qualifiers data can be found in [this spreadsheet](https://docs.google.com/spreadsheets/d/e/2PACX-1vSNjTR6oXcjyCEaNEu-h0EabC8ZSYUMVhjEUYvZsfU78JXOrUpoOc_lIDinRe2gWEDbDCkTdvr1VCn9/pubhtml).

### Further analysis

Here are some other features and comparisons of seeding methods to keep in mind, especially for tournaments using methods not listed on this page.

- Some seeding methods are not very sensitive overall to individual teams' scores, while others may be changed more dramatically.

>   - For example, Sum of Placements yields the same placements no matter how large the differences between team scores are, but a particularly high or low score in Percent Difference may cause all other `map points` to be unusually close together. This is particularly important for modes like osu!taiko or osu!mania where scores are typically very close to the maximum possible; historically, some tournaments such as [SOFT](/wiki/Tournaments/SOFT) have instead used a "piecewise-linear" Percent Difference method by grouping scores by quartile.

Reviewer 2: ```suggestion
  - For example, sum of placements yields the same placements no matter how large the differences between team scores are, but a particularly high or low score in percent difference may cause all other `map points` to be unusually close together. This is particularly important for modes like osu!taiko or osu!mania where scores are typically very close to the maximum possible; historically, some tournaments such as [SOFT](/wiki/Tournaments/SOFT) have instead used a "piecewise-linear" percent difference method by grouping scores by quartile.
```

>   - For another kind of example, the Z-sum and Z-percentile seeding methods have the feature that very close scores will receive very close point values for seeding (so if there is a cluster of many nearby scores, the highest and lowest scores of those clusters will be considered similarly in seeding), which is behaviour that does not occur for Sum of Placements.

Reviewer 2: ```suggestion
  - For another kind of example, the Z-sum and Z-percentile seeding methods have the feature that very close scores will receive very close point values for seeding (i.e. if there is a cluster of many nearby scores, the highest and lowest scores of those clusters will be considered similarly in seeding), which is behaviour that does not occur for the sum of placements method.
```

> - Some seeding methods benefit specialists, while others benefit consistent performances. For example, Zipf's Law will give a large boost in seeding to teams with the highest or near-highest rank on a map compared to Sum of Placements. And because of the shape of the normal distribution, Z-sum will give a large boost for extremely high scores (and a large reduction for extremely low scores) compared to Z-percentile; thus, Z-percentile tends to favour all-rounders over specialists compared to Z-sum.

Reviewer 2: ```suggestion
- Some seeding methods benefit specialists, while others benefit consistent performances. For example, Zipf's law will give a large boost in seeding to teams with the highest or near-highest rank on a map compared to sum of placements. Similarly, because of the shape of the normal distribution, Z-sum will give a large boost for extremely high scores (and a large reduction for extremely low scores) compared to Z-percentile. Z-percentile therefore tends to favour all-rounders over specialists compared to Z-sum.
```

- Some seeding methods may cause certain maps to contribute more to seeding differences than others.

>   - In particular, weighted seeding methods are designed to allow some maps to factor more into overall seeding than others – this allows for balancing of importance across different skill sets without needing to include extra maps of the same type in a pool.

Reviewer: ```suggestion
  - In particular, weighted seeding methods are designed to allow some maps to factor more into overall seeding than others – this allows for balancing of importance across different skill sets without needing to include extra maps of the same type in a mappool.
```

>   - In particular, weighted seeding methods are designed to allow some maps to factor more into overall seeding than others – this allows for balancing of importance across different skill sets without needing to include extra maps of the same type in a mappool.

Reviewer 2: ```suggestion
  - In particular, weighted seeding methods are designed to allow some maps to factor more into overall seeding than others — this allows for balancing of importance across different skill sets without needing to include extra maps of the same type in a mappool.
```

>   - For a less direct example, seeding by average score may cause relative performance on easier maps to affect seeding more than relative performance on harder maps (if it is easier to increase score by a fixed amount on the easier map than the harder one). Along the same lines, one particularly high team score on a map may mean that seeding by Percent Max or Percent Difference will give that one team a large boost while distinguishing the other teams on the map less.

Reviewer 2: ```suggestion
  - For a less direct example, seeding by average score may cause relative performance on easier maps to affect seeding more than relative performance on harder maps (if it is easier to increase score by a fixed amount on the easier map than the harder one). Along the same lines, one particularly high team score on a map may mean that seeding by percent max or percent difference will give that one team a large boost while distinguishing the other teams on the map less.
```

## Official Support

If you’re seeking [official support](/wiki/Tournaments/Official_support) for your tournament and you are using the formulas or methods as described on this page exactly as they are written, you may refer to the name as described in the [seeding methods](#seeding-methods) section instead of specifying the exact formula. However, should there be any difference or change from the methods described here, you will be required to provide the full formula in your tournament’s ruleset. If you are using a different seeding method not found on this page, make sure to specify the exact formulas and/or method used.

As always, a third party should be able to get the same results without needing additional information.

## Mini-glossary

In the explanations below, we use the following terms:

- A **team** is a single entity participating in the tournament. For example, the 6-8 players from each country in [OWC](/wiki/Tournaments/OWC) are considered one team, while each player in the [Lazer Grand Arena](/wiki/Tournaments/LGA/2024) is considered their own team.

> - A **score** is a numerical value used for determining a team’s performance on a given map. For example, in a 3v3 team tournament, a score typically refers to the sum of the three team members’ scores, rather than those three numbers separately. Some tournaments may allow teams multiple runs through a qualifier mappool; in those cases, the team’s score is often either the highest or the average of those runs.

Reviewer: Shouldn't it be lowercase here since it doesn't refer to the name of the stage but rather a qualifier mappool in general?

- A team’s **seed** is a number indicating their overall ranking among all teams. Typically, after one of the seeding methods below is used, the top N teams (that is, the teams seeded 1, 2, …, N) are placed into an elimination bracket or other competition format, with their locations being determined by their seeds.
- **Normalisation** is the process of dividing a team’s score by any mod multipliers on the map. Typically, this is done to avoid artificial inflation of scores on modded maps.

> ## Seeding methods

Author: All seeding methods have each word capitalised, all other headers are sentence case.

<!-- NOTE FOR WIKI EDITORS: Please do not add or remove seeding methods without prior approval of the Tournament Committee -->

Each subsection below explains, both in words and in a mathematical formula, a different seeding method commonly found in osu! tournaments. In all formulas, `m` refers to the number of maps in the qualifier mappool.

### Average Score

This seeding method averages each team’s scores on all maps. Typically, scores are first normalised, meaning that they are divided by the [mod multipliers](/wiki/Gameplay/Game_modifier/Mod_multiplier) specified on the map, before averaging. (For example, scores on an osu! map played with Hard Rock in ScoreV2 would be divided by 1.1.) Teams are then seeded in order of this average, with the highest average receiving seed 1.

The average score for a team is given by

![Average Score formula](img/avgscore.png)[^mappool-size]

> If any of the maps are played with [Free Mod](/wiki/Gameplay/Game_modifier#free-mod), it must be explicitly specified whether each individual player’s score is being normalised by the mods they used before summing, or if there is an overall normalisation.

Reviewer 4: This is meant as like "blanket" normalisation, so dividing by a common value for all scores regardless of mod.

> If any of the maps are played with [Free Mod](/wiki/Gameplay/Game_modifier#free-mod), it must be explicitly specified whether each individual player’s score is being normalised by the mods they used before summing, or if there is an overall normalisation.

Reviewer 3: Yep, one example would be a map played with DT where HD is optionally allowed. "Individual normalisation" divides out all mod multipliers (effectively nullifying the HD multiplier for those players), while "overall normalisation" would divide everyone's score only by the DT multiplier. Both options are sensible and I don't think there's any consensus in the scene for now (?).

> If any of the maps are played with [Free Mod](/wiki/Gameplay/Game_modifier#free-mod), it must be explicitly specified whether each individual player’s score is being normalised by the mods they used before summing, or if there is an overall normalisation.

Reviewer 5: Okay. Might be good to define overall normalization somewhere if not done already.

> If any of the maps are played with [Free Mod](/wiki/Gameplay/Game_modifier#free-mod), it must be explicitly specified whether each individual player’s score is being normalised by the mods they used before summing, or if there is an overall normalisation.

Reviewer 3: It's definitely a bit hard to phrase this... maybe something like the following?
```suggestion
If any of the maps are played with [Free Mod](/wiki/Gameplay/Game_modifier#free-mod), it must be explicitly specified whether each individual player’s score on the map is normalised by the mods they use (effectively nullifying multipliers of any additional mods taken), or if there is an overall normalisation (where all scores are divided by only the globally required mod multipliers).
```

> If any of the maps are played with [Free Mod](/wiki/Gameplay/Game_modifier#free-mod), it must be explicitly specified whether each individual player’s score is being normalised by the mods they used before summing, or if there is an overall normalisation.

Reviewer 5: Maybe we remove the stuff in parentheses and add a footnote (i.e. a `[1]` people can click to see more details on the bottom) which then explains what you wrote in parentheses / explains the difference between these normalisations? Alternatively we can define normalisation above with the other definitions.

### Sum of Placements

This seeding method ranks all scores on each map played from highest to lowest, with the highest-scoring team receiving a rank of 1, the next receiving 2, and so on. A team’s overall seeding is then determined by the sum of these numbers (called placements); the lowest sum receives seed 1, the next lowest receives seed 2, and so on.

The sum of placements for a team is given by

![Sum of Placements formula](img/golf.png)[^mappool-size]

Instead of a sum, the placements may also be averaged, which yields the same overall seeding. Note that if two teams both receive the highest score on a map, their map placements are both 1 (rather than both being 2); more explicitly, a team’s `Map placement` on a map is the number of teams who score higher than them plus one.

### Percent Max

> This seeding method assigns each team some number of points between 0 and 1 on each map by dividing their score by the highest score achieved. Seeding is then determined by the sum of these map points, with the highest sum receiving seed 1.

Author: That is trivial if you conider that the highest score on the map divided by the highest score on that map is always going to be equal to 1. Doesn't need further elaboration in my opinion.

The percent max sum for a team is given by

![Percent Maximum formula](img/percentmax.png)[^mappool-size]

### Percent Difference

This seeding method is almost identical to Percent Max, except that it assigns the lowest score on each map 0, the highest score on each map 1, and all others linearly in between. 

The percent difference sum for a team is given by

![Percent Difference formula](img/percentdiff.png)[^mappool-size]

### Z-Sum

This seeding method first computes the average and standard deviation of team scores on each map. Each team then receives a “z-score” for each map, which is the number of standard deviations above or below the mean that they scored. (0 is an average z-score, and z-scores are typically between -3 and 3.) Seeding is then determined by the sum of these z-scores, with the highest sum receiving seed 1.

> Note that tournaments in the past have often used the term “Z-Sum” for what is instead called the “Z-Percentile” seeding method below.[^z-sum] The Z-Sum seeding method should result in roughly half the teams having a negative total sum of z-scores and half having a positive sum. The total sum over all maps should be exactly 0.

Author: The [^Z-Sum] footnote does this no? See the preview to view what that looks like on the wiki.

The z-sum for a team is given by

![Z-Sum formula](img/zsum.png)[^mappool-size]

where typically the sample standard deviation is used (rather than the population standard deviation).

### Z-Percentile

> This seeding method is almost identical to Z-Sum, except that instead of assigning teams a z-score, it assigns teams the corresponding percentile with respect to the normal distribution. In other words, Z-Percentile estimates the percentage of teams beaten on each map by fitting scores to a bell curve, and each team will receive a value between 0 and 1 on each map played. Seeding is then determined by the sum of these values, with the highest sum receiving seed 1.

Author: There's a document linked under "Comparing the most commonly used methods" that showcases all the different methods and their effects on Qualifier results.

## Differences between methods

There are many factors that go into choosing a seeding method for a tournament. First of all, the following principles are important to keep in mind:

- **Lack of qualifiers**: All of the seeding methods above assume that teams all play through the same qualifiers pool. If qualifiers are not played, then it may make sense instead to seed by rank or [BWS](/wiki/Tournaments/Badge-weighted_seeding). (For example, a 2v2 tournament with team size 4 may seed teams by the sum of their two best BWS ranks, with the smallest sum being seeded 1st.)
  - Note that some tournaments may begin head-to-head competition by placing teams into a group stage or Swiss-system tournament instead of a standard elimination bracket. However, these other systems still require specification on how the teams are placed into matchups, so often some method of seeding is still required (either one of the ones mentioned in this page, or random drawings).

> - **Scoring system**: Most of the seeding methods above are typically used when tournaments are played with the ScoreV2 scoring system. Thus, extra care should be taken when seeding tournaments that use ScoreV1, accuracy, or other methods of scoring. For example, a map played with ScoreV1 may have high outliers from well-performing teams, so it may be less advised to seed by Z-Percentile or Percent Maximum.

Reviewer 5: ```suggestion
- **Scoring system**: Most of the seeding methods above are typically used when tournaments are played with the ScoreV2 scoring system. Thus, extra care should be taken when seeding tournaments that use ScoreV1, accuracy, or other methods of scoring. For example, a map played with ScoreV1 may have high outliers from well-performing teams, so it may be less appropriate to seed by Z-Percentile or Percent Maximum.
```
